File: src/main/java/jadx/plugins/example/HookType.java
package jadx.plugins.example;

public enum HookType {
	BEFORE,
	AFTER,
	INSTEAD;

	public String asString(CodegenLanguage codegen) {
		switch (codegen) {
			case JAVA: switch (this) {
				case BEFORE:
					return "PreHook";
				case AFTER:
					return "Hook";
				case INSTEAD:
					return "InsteadHook";
			}
			case KOTLIN: return name().toLowerCase();
			default: return null;
		}
	}
}

File: src/main/java/com/github/canny1913/HookType.kt
package com.github.canny1913

enum class HookType {
	BEFORE,
	AFTER,
	INSTEAD;

	fun asString(codegen: CodegenLanguage): String {
		return when (codegen) {
			CodegenLanguage.JAVA -> {
				when (this) {
					BEFORE -> "PreHook"
					AFTER -> "Hook"
					INSTEAD -> "InsteadHook"
				}
			}
			CodegenLanguage.KOTLIN -> this.name.lowercase()

		}
	}
}
